src/AST/CommonAST.cs

```diff
@@ -142,6 +144,62 @@ public enum SpanFilterCombination
     All
 }
 
+/// <summary>
+/// Project operation node representing both KQL's project operator and TraceQL's select operation
+/// Projects specified fields/columns from the data, optionally with aliases
+/// 
+/// Level 1 Support (Current):
+/// - Simple field references
+/// - Aliasing  
+/// - Basic arithmetic operations (+, -, *, /)
+/// - Simple function calls (string, math, conversion)
+/// 
+/// Level 2 Support (Future TODO):
+/// - Complex nested expressions
+/// - Conditional expressions (case)
+/// - Advanced function calls
+/// </summary>
+public class ProjectNode : OperationNode
+{
+    public override NodeKind NodeKind => NodeKind.Project;
+    
+    /// <summary>
+    /// List of field projections to include in the output
+    /// No wildcard (*) support - if all fields needed, project operation should be omitted
+    /// </summary>
+    public List<ProjectionExpression> Projections { get; set; } = new List<ProjectionExpression>();
+    
+    /// <summary>
+    /// Keyword for language-specific syntax ('project' for KQL, 'select' for TraceQL)
+    /// </summary>
+    public string? Keyword { get; set; }
```

Comment:
Didn't we ditch the keyword for some of the other operators as it was unnecessary?